modules/platforms/cpp/core-test/src/cluster_node_test.cpp

```diff
@@ -0,0 +1,261 @@
+    BOOST_REQUIRE(nodes[1].IsAttributeSet(ATTR_CLIENT_MODE));
+    bool isClient0 = nodes[0].GetAttribute<bool>(ATTR_CLIENT_MODE);
+    bool isClient1 = nodes[1].GetAttribute<bool>(ATTR_CLIENT_MODE);
+    BOOST_REQUIRE(!isClient0 && isClient1);
+
+    const std::string ATTR_GRID_NAME = "org.apache.ignite.ignite.name";
+    BOOST_REQUIRE(nodes[0].IsAttributeSet(ATTR_GRID_NAME));
+    std::string gridName = nodes[0].GetAttribute<std::string>(ATTR_GRID_NAME);
+    BOOST_REQUIRE(!gridName.empty());
+
+
+    const std::string ATTR_PHY_RAM = "org.apache.ignite.phy.ram";
+    BOOST_REQUIRE(nodes[0].IsAttributeSet(ATTR_PHY_RAM));
+    int64_t phyRam = nodes[0].GetAttribute<int64_t>(ATTR_PHY_RAM);
+    BOOST_REQUIRE(phyRam > 0);
+}
+
+BOOST_AUTO_TEST_CASE(IgniteGetAttributes)
+{
+    IgniteCluster cluster = server.GetCluster();
+
+    BOOST_REQUIRE(cluster.IsActive());
+
+    ClusterGroup group = cluster.AsClusterGroup();
+
+    std::vector<ClusterNode> nodes = group.GetNodes();
+
+    BOOST_REQUIRE(nodes.size() == 2);
+
+    std::vector<std::string> attrs = nodes[0].GetAttributes();
+
+    BOOST_REQUIRE(!attrs.empty());
+}
+
+BOOST_AUTO_TEST_CASE(IgniteGetConsistentId)
+{
+    IgniteCluster cluster = server.GetCluster();
+
+    BOOST_REQUIRE(cluster.IsActive());
+
+    ClusterGroup group = cluster.AsClusterGroup();
+
+    std::vector<ClusterNode> nodes = group.GetNodes();
+
+    BOOST_REQUIRE(nodes.size() == 2);
+
+    std::string consistentId0 = nodes[0].GetConsistentId();
+    std::string consistentId1 = nodes[1].GetConsistentId();
+
+    BOOST_REQUIRE(!consistentId0.empty());
+    BOOST_REQUIRE(!consistentId1.empty());
+}
+
+BOOST_AUTO_TEST_CASE(IgniteGetHostNames)
+{
+    IgniteCluster cluster = server.GetCluster();
+
+    BOOST_REQUIRE(cluster.IsActive());
+
+    ClusterGroup group = cluster.AsClusterGroup();
+
+    std::vector<ClusterNode> nodes = group.GetNodes();
+
+    BOOST_REQUIRE(nodes.size() == 2);
+
+    std::vector<std::string> hosts = nodes[0].GetHostNames();
+
+    BOOST_REQUIRE(hosts.empty());
+}
+
+BOOST_AUTO_TEST_CASE(IgniteGetId)
+{
+    IgniteCluster cluster = server.GetCluster();
+
+    BOOST_REQUIRE(cluster.IsActive());
+
+    ClusterGroup group = cluster.AsClusterGroup();
+
+    std::vector<ClusterNode> nodes = group.GetNodes();
+
+    BOOST_REQUIRE(nodes.size() == 2);
+
+    Guid id1 = nodes[0].GetId();
+    Guid id2 = nodes[1].GetId();
+
+    BOOST_REQUIRE(id1 != Guid() && id2 != Guid() && id1 != id2);
+}
+
+BOOST_AUTO_TEST_CASE(IgniteIsClientDaemonLocal)
+{
+    IgniteCluster cluster = server.GetCluster();
+
+    BOOST_REQUIRE(cluster.IsActive());
+
+    ClusterGroup group = cluster.AsClusterGroup();
+
+    std::vector<ClusterNode> nodes = group.GetNodes();
+
+    BOOST_REQUIRE(nodes.size() == 2);
+
+    bool isClient = nodes[0].IsClient();
+    bool isDaemon = nodes[0].IsDaemon();
+    bool isLocal = nodes[0].IsLocal();
+
+    BOOST_REQUIRE(!isClient && !isDaemon && isLocal);
```

Comment:
It would be better to have separate assertions so we can understand which condition has failed